packages/astro/src/runtime/server/endpoint.ts

```diff
@@ -23,38 +11,34 @@ export async function renderEndpoint(
 ) {
 	const { request, url } = context;
 
-	const chosenMethod = request.method?.toUpperCase();
-	const handler = getHandlerFromModule(mod, chosenMethod);
-	if (!ssr && ssr === false && chosenMethod && chosenMethod !== 'GET') {
+	const method = request.method.toUpperCase();
+	// use the exact match on `method`, fallback to ALL
+	const handler = mod[method] ?? mod['ALL'];
+	if (!ssr && ssr === false && method !== 'GET') {
 		logger.warn(
-			null,
+			"router",
 			`${url.pathname} ${bold(
-				chosenMethod
+				method
 			)} requests are not available for a static site. Update your config to \`output: 'server'\` or \`output: 'hybrid'\` to enable.`
 		);
 	}
-	if (!handler || typeof handler !== 'function') {
+	if (typeof handler !== 'function') {
+		logger.warn(
+			'router',
+			`No API Route handler exists for the method "${method}" for the route ${url.pathname}.\n` +
+			`Found handlers: ${Object.keys(mod).map(exp => JSON.stringify(exp)).join(', ')}\n` +
+			('all' in mod ? `One of the exported handlers is "all" (lowercase), did you mean to export 'ALL'?\n` : '')
```

Comment:
Just wondering why `all` is the only case to handle, and not any of the other verbs?

I would assume that, if the user reads "get" in the `Found handlers:` log, it should be enough.

Reply:
The most recent user to run into this was exporting "all". I agree this message should expand to "get", "post" depending what the current request's method is.